### DISCUSSION TITLE
Roles and Premissions

### DISCUSSION THREAD

**Discussion by MasterAdmin2** [filamentphp/filament#9988] [Ideas]
### Introduction:
Managing roles and permissions is a crucial aspect of building robust and secure web applications. While there are several third-party packages available for Laravel that offer these features, it's equally empowering to implement a custom solution, avoiding unnecessary dependencies. This approach not only promotes efficiency but also provides a deep understanding of the inner workings of your application.

### Guiding Principles:
Simplicity and Efficiency:
We aim for a straightforward and efficient implementation that aligns seamlessly with Laravel's eloquent syntax.

Customization and Control:
Crafting our own roles and permissions system allows for complete customization, ensuring that the solution perfectly fits the unique requirements of the application.

Integration with Laravel's Native Features:
Leveraging Laravel's built-in features, such as eloquent relationships and middleware, enhances the cohesion of our implementation.

### Benefits:
Custom Fit:
Tailor the roles and permissions system to the exact needs of your application, avoiding unnecessary bloat.

Deeper Understanding:
Gain a profound understanding of Laravel's capabilities by building a core feature from scratch, empowering developers with in-depth knowledge.

Performance:
Minimize the overhead associated with third-party packages, potentially resulting in a more performant application.

**Comment by leandrocfe** [comment]
https://filamentphp.com/plugins/bezhansalleh-shield

?

**Comment by MasterAdmin2** [comment]
> https://filamentphp.com/plugins/bezhansalleh-shield
> 
> ?

yes that's fine, but i was looking like, they gave inbuilt addon feature from filament on its own without any third-party apps
